src/consensus/pbft_chain.cpp

```diff
@@ -14,18 +14,22 @@ PbftBlock::PbftBlock(bytes const& b) : PbftBlock(dev::RLP(b)) {}
 PbftBlock::PbftBlock(dev::RLP const& r) {
   dev::RLP const rlp(r);
   if (!rlp.isList()) throw std::invalid_argument("PBFT RLP must be a list");
-  prev_block_hash_ = rlp[0].toHash<blk_hash_t>();
-  dag_block_hash_as_pivot_ = rlp[1].toHash<blk_hash_t>();
-  period_ = rlp[2].toInt<uint64_t>();
-  timestamp_ = rlp[3].toInt<uint64_t>();
-  signature_ = rlp[4].toHash<sig_t>();
```

Comment:
btw no need to change that, if you accessing RLP elements in ascending order it is exactly same as iterator

Reply:
Internally RLP does not have random access, so accessing by index actually internally makes it iterate multiple time

Reply:
yes, but it is not true if you doing it in ascending order, because it is doing same thing as you, https://github.com/Taraxa-project/taraxa-aleth/blob/taraxa_master/libdevcore/RLP.cpp#L66 you can see, it has internal iterator `m_lastEnd`, so no need to rewrite anything if your are accessing something like 1,2,3... or even 1,4,12 ...

Reply:
I see. Thanks

Reply:
you do not need to change it, just letting you know